lib/ansible/modules/storage/glusterfs/gluster_volume.py

```diff
@@ -354,6 +366,50 @@ def add_bricks(name, new_bricks, stripe, replica, force):
     run_gluster(args)
 
 
+def remove_bricks(name, removed_bricks, force):
+    # max-tries=12 with default_interval=10 secs
+    max_tries = 12
+    retries = 0
+    success = False
+    args = ['volume', 'remove-brick', name]
+    args.extend(removed_bricks)
+    # create a copy of args to use for commit operation
+    args_c = args[:]
+    args.append('start')
+    run_gluster(args)
+    # remove-brick operation needs to be followed by commit operation.
+    if not force:
+        module.fail_json(msg="Force option is mandatory.")
+    else:
+        while retries < max_tries:
+            last_brick = removed_bricks[-1]
+            out = run_gluster(['volume', 'remove-brick', name, last_brick, 'status'])
+            for row in out.split('\n')[1:]:
+                if 'completed' in row:
+                    # remove-brick successful, call commit operation.
+                    args_c.append('commit')
+                    out = run_gluster(args_c)
+                    success = True
+                    break
+                else:
+                    time.sleep(10)
+            if success:
+                break
+            retries += 1
+        if not success:
+            # remove-brick still in process, needs to be committed after completion.
+            module.fail_json(msg="Exceeded number of tries, check remove-brick status.\n"
+                                 "Commit operation needs to be followed.")
+
+
+def reduce_config(name, removed_bricks, replicas, force):
+    args = ['volume', 'remove-brick', name, 'replica', replicas]
+    args.extend(removed_bricks)
+    if force:
+        args.append('force')
+    run_gluster(args)
```

Comment:
When reducing the replica count, the force option is mandatory. This will fail if force is not set.
Also for remove brick can you check there are not pending heals. 

If we are removing a brick from a volume which has pending heals and if that brick had the only good copy, we have data loss.

Reply:
@sac thanks for pointing it out. I have added checks for self-heal status before executing the remove-brick operation when reducing the cluster configuration.
Thanks.